scripts/wasmtestreport.py

```diff
@@ -622,6 +611,58 @@ def compare_test_results(file1, file2):
     
     return (status, new_failures)
 
+# ----------------------------------------------------------------------
+# Function: run_subprocess
+#
+# Purpose:
+#   Wrapper for subprocess.run with debug logging. 
+#   Improves visibility during unit test execution, and replaces unsafe os.chdir usage by allowing a cwd argument.
+#
+# Variables:
+# - Input:
+#   cmd     : Command to execute (e.g., ["gcc", "main.c"]).
+#   label   : Label used to tag debug log outputs.
+#   cwd     : Working directory for the command.
+#   shell   : Whether to run the command in a shell.
+#   timeout : Timeout for command execution.
+#
+# - Output:
+#   The result of subprocess.run()
+#
+# Raises:
+#   subprocess.TimeoutExpired - If the command exceeds the timeout.
+#   Exception - For all other unexpected execution errors.
+# ----------------------------------------------------------------------
+def run_subprocess(cmd, label="", cwd=None, shell=False, timeout=None):
```

Comment:
Could there be a bit of a guard rail here to prevent strange errors from calling this in the future? Something like expressly checking that if Shell=False, cmd must be a list whereas if Shell=True, cmd must be a string.  I know right now we're only doing one, but it would protect us later if the other was added.